### PR TITLE
feat: unread messages and mark them as read

### PR DESCRIPTION
**Pull request by likeri29**
## Describe your changes
- I added the functionality to see how many messages are unread
- I added the possibility to mark messages as read
- Changed the response for retrieving all chats (now it includes the unredMessagesCount and the last message instead of all)
- I added the new route for creating the chat with admin

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-259)
  
### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/chat/chat.service.ts
Comment (djanbolotov) [inline src/modules/chat/chat.service.ts:1]: Sort order of allChats (first is last message of lastMessages in all chats) and chatById (first last message of messages in chat).

File: src/modules/chat/chat.service.ts [hunk 1]
```diff
@@ -39,9 +55,31 @@ export class ChatService {
     }
 
     const userChats = user.chatRooms;
+    const EQUAL_SORT_ORDER = 0;
+
+    const sortedUserChats = await Promise.all(
```
Comment (lyudad) [inline src/modules/chat/chat.service.ts]: why you dont sort it in the query?

Comment (lyudad) [reply src/modules/chat/chat.service.ts]: but you need you use query builder, I'm not sure it possible with repository

File: src/modules/events/events.gateway.ts [hunk 2]
```diff
@@ -90,23 +95,36 @@ export class EventsGateway
...
+  //   chatRoom.participants.forEach((participant) => {
+  //     client.join(participant.id);
+  //     this.server.to(participant.id).emit('newChat', chatRoom);
+  //   });
+
+  //   this.server.to(chatRoom.id).emit('newChat', chatRoom);
+
+  //   return chatRoom;
+  // }
+
```
Comment (lyudad) [inline src/modules/events/events.gateway.ts]: delete

File: src/modules/chat/chat.service.ts (hunk 1 above)
Comment (likeri29) [reply src/modules/chat/chat.service.ts]: Agree. Fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**